src/NRedisStack/Json/IJsonCommandsAsync.cs

```diff
@@ -123,10 +124,11 @@ public interface IJsonCommandsAsync
     /// </summary>
     /// <param name="key">The key to retrieve</param>
     /// <param name="path">The path to retrieve</param>
+    /// <param name="serializerOptions">Json serializer options to use for deserialization.</param>
     /// <typeparam name="T">The type retrieved</typeparam>
     /// <returns>The object requested</returns>
     /// <remarks><seealso href="https://redis.io/commands/json.get"/></remarks>
-    Task<T?> GetAsync<T>(RedisKey key, string path = "$");
+    Task<T?> GetAsync<T>(RedisKey key, string path = "$", JsonSerializerOptions? serializerOptions = default);
```

Comment:
@MichaelDeutschCoding

Reply:
I don't really have the context to know which of those options makes the most sense; this is my first contribution to the code base. I'm happy to implement any of those approaches, if you can just give me the direction you'd like to take.

Reply:
@slorello89 
To create another signature where serializerOptions is not optional, we will have to do it like this:
```csharp
T? Get<T>(RedisKey key, JsonSerializerOptions serializerOptions, string path = "$");
```
What changes the order of the original function (path after key), I think it's better to leave serializerOptions optional.
WDYT?

Reply:
You would just reset this line, and add a new method with:

```cs
Task<T?> GetAsync<T>(RedisKey key, string path = "$", JsonSerializerOptions serializerOptions);
```

as the signature.

Reply:
@slorello89 I didn't understand how this is possible, an optional variable should come at the end

Reply:
I don't know how bad it would be breaking binary/reflection users by changing the method signature, but personally it would feel weird to me to add an additional separate method with a mandatory JsonSerializerOptions, when that parameter is itself optional in the [JsonSerializer method](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=net-7.0#system-text-json-jsonserializer-deserialize-1%28system-string-system-text-json-jsonserializeroptions%29) we're calling.

If the version number is bumped, does that provide enough "warning" if it does indeed break anyone's use flow?

Reply:
Honestly, I didn't understand why it breaks.
If the variable we added is optional, anyone who used the function without `JsonSerializerOptions` can continue to use it in exactly the same way because the variable is optional.
@slorello89 what do you say?